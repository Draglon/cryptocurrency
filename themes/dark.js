export default {
  // React Native Elements
  Button: {
    containerStyle: {
      width: 150,
    },
    titleStyle: {
      color: '#00f',
    },
  },
  Icon: {
    type: 'font-awesome',
    size: 24,
  },

  // Custom Elements
  IconClose: {
    containerStyle: {
      backgroundColor: '#ccc',
      color: '#f00',
      position: 'absolute',
      top: 10,
      right: 10,
      width: 30,
      height: 30,
      lineHeight: 30,
      textAlign: 'center',
    },
  }
};
