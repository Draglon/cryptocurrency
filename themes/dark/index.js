import { COLORS } from '../../constants/variables';

import Header from './Header';
import MenuButton from './MenuButton';
import {
  Sidebar, SidebarHeader, SidebarBody, SidebarWrapper,
} from './Sidebar';
import { Menu, MenuItem, MenuItemIcon } from './Menu';
import { Themes, ThemesItem } from './Themes';
import { Languages, LanguagesItem } from './Languages';
import { Page, PageBody, PageWrapper } from './Page';
import { MySets, MySetsEmpty } from './MySets';
import { CreateSet } from './CreateSet';
import Logo from './Logo';
import BackLink from './BackLink';
import ButtonPrimary from './ButtonPrimary';
import ButtonClose from './ButtonClose';

export default {
  // ================= React Native Elements =================
  Divider: {
    backgroundColor: COLORS.white,
    height: 2,
  },
  Text: {
    h1Style: {
      color: COLORS.white,
      marginBottom: 15,
    },
    h2Style: {
      color: COLORS.white,
      marginBottom: 15,
    },
    h3Style: {
      color: COLORS.white,
      marginBottom: 15,
    },
    h4Style: {
      color: COLORS.white,
      marginBottom: 12,
    },
    style: {
      color: COLORS.grayLight,
      marginBottom: 10,
    },
  },
  Icon: {
    type: 'font-awesome',
    size: 18,
  },

  // ================ Custom Elements ========================
  // Header
  Header,
  MenuButton,
  // Pages
  Page,
  PageBody,
  PageWrapper,
  // Sidebar
  Sidebar,
  SidebarHeader,
  SidebarBody,
  SidebarWrapper,
  Menu,
  MenuItem,
  MenuItemIcon,
  Themes,
  ThemesItem,
  Languages,
  LanguagesItem,
  // My sets - page
  MySets,
  MySetsEmpty,
  // Create set - page
  CreateSet,
  // Shared
  Logo,
  BackLink,
  ButtonPrimary,
  ButtonClose,
};
