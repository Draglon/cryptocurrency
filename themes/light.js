export default {
  // React Native Elements
  Button: {
    containerStyle: {
      width: 100,
    },
    titleStyle: {
      color: '#fff',
    },
  },
  Icon: {
    type: 'font-awesome',
    size: 24,
  },

  // Custom Elements
  IconClose: {
    containerStyle: {
      backgroundColor: '#ccc',
      color: '#f00',
      position: 'absolute',
      top: 10,
      right: 10,
      width: 30,
      height: 30,
      lineHeight: 30,
      textAlign: 'center',
    },
  },
};
