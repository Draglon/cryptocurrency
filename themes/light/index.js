import { COLORS } from '../../constants/variables';

import Header from './Header';
import MenuButton from './MenuButton';
import {
  Sidebar, SidebarHeader, SidebarBody, SidebarWrapper,
} from './Sidebar';
import { Menu, MenuTitle, MenuItem } from './Menu';
import { Themes, ThemesItem } from './Themes';
import { Languages, LanguagesItem } from './Languages';
import { Page, PageBody, PageWrapper } from './Page';
import { MySets, MySetsEmpty } from './MySets';
import { CreateSet } from './CreateSet';
import Logo from './Logo';
import BackLink from './BackLink';
import ButtonClose from './ButtonClose';
import ButtonPrimary from './ButtonPrimary';
import ButtonCheckbox from './ButtonCheckbox';

export default {
  // ================= React Native Elements =================
  Divider: {
    style: {
      backgroundColor: COLORS.gray,
      height: 2,
    },
  },
  Text: {
    h1Style: {
      color: COLORS.gray,
      marginBottom: 15,
    },
    h2Style: {
      color: COLORS.gray,
      marginBottom: 15,
    },
    h3Style: {
      color: COLORS.gray,
      marginBottom: 15,
    },
    h4Style: {
      color: COLORS.gray,
      marginBottom: 15,
    },
    style: {
      color: COLORS.grayLight,
      marginBottom: 10,
    },
  },
  Icon: {
    type: 'font-awesome',
    size: 18,
  },

  // ================ Custom Elements ========================
  // Header
  Header,
  MenuButton,
  // Pages
  Page,
  PageBody,
  PageWrapper,
  // Sidebar
  Sidebar,
  SidebarHeader,
  SidebarBody,
  SidebarWrapper,
  Menu,
  MenuTitle,
  MenuItem,
  Themes,
  ThemesItem,
  Languages,
  LanguagesItem,
  // My sets - page
  MySets,
  MySetsEmpty,
  // Create set - page
  CreateSet,
  // Shared
  Logo,
  BackLink,
  ButtonClose,
  ButtonPrimary,
  ButtonCheckbox,
};
