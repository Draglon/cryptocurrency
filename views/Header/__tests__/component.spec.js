import React from 'react';
import { shallow } from 'enzyme';

import { THEMES } from '../../../themes';
import HeaderComponent from '../component';

describe('Header component matches snapshot', () => {
  const defaultProps = {
    navigation: {
      navigate: jest.fn(),
    },
    theme: {
      Header: {
        ...THEMES.light.Header,
      },
    },
  };

  it('with default props', () => {
    const wrapper = shallow(<HeaderComponent {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('with dark theme', () => {
    const props = {
      ...defaultProps,
      theme: {
        Header: {
          ...THEMES.dark.Header,
        },
      },
    };
    const wrapper = shallow(<HeaderComponent {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
