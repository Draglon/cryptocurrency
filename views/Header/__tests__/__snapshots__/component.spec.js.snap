// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot with dark theme 1`] = `ShallowWrapper {}`;

exports[`Header component matches snapshot with default props 1`] = `ShallowWrapper {}`;
